data/docs/gcp-monitoring/gke-tracing.mdx

```diff
@@ -0,0 +1,1190 @@
+  - image: mongo:latest
+    name: mongo
+    ports:
+    - name: mongo
+      containerPort: 27017
+  - image: signoz/sample-flask-app:latest
+    name: sample-app
+    ports:
+    - name: flask
+      containerPort: 5002
+    env:
+    - name: MONGO_HOST
+      value: localhost
+    - name: OTEL_RESOURCE_ATTRIBUTES
+      value: service.name=sample-app
+    - name: OTEL_EXPORTER_OTLP_ENDPOINT
+      value: http://kubelet-otel-k8s-infra-otel-agent.signoz.svc.cluster.local:4317
+    - name: OTEL_EXPORTER_OTLP_PROTOCOL
+      value: grpc
+  dnsPolicy: ClusterFirst
+  restartPolicy: Always 
+``` 
+ 
+### Deployment & Expose port
+
+```sh 
+kubectl apply -f sample-flask-app.yaml 
+```
+One can expose the application via NodePort or LoadBalancer as service. 
+ 
+```sh 
+kubectl expose pod --port=5002 --name=sample-flask-svc –type NodePort 
+```
+
+
+## Update OTel config map configuration
+
+You need to add IP address/domain name of self hosted SigNoz at OTel config map:
+
+Edit the config map of the OTel agent
+
+```yaml
+apiVersion: v1
+data:
+  otel-agent-config.yaml: |2-
+
+    exporters:
+      otlp:
+        endpoint: "<IP ADDRESS>:4317" 
+        tls:
+           insecure: true 
+```
+
+### Visualize tracing in SigNoz
+
+After editing the config map, and once the application restarts, upon performing any action on the application UI/endpoint, it should send traces that should be visible over the self hosted SigNoz: 
+
+<figure data-zoomable align="left">
+    <img
+        src="/img/docs/gcp-monitoring/gke-tracing/self-hosted-application-level.webp"
+        alt="Application-level traces in SigNoz"
+    />
+<figcaption>
+<i>
+Application-level traces in SigNoz
+</i>
+</figcaption>
+</figure>
+
+## Troubleshooting
+
+If you encounter any issues while setting up logging and metrics for your GKE cluster, follow these troubleshooting steps:
+
+<Tabs>
+<TabItem value="self-hosted-on-gke-cluster" label="For Self-hosted SigNoz in GKE Cluster" default>
+
+1. Confirm that the necessary Kubernetes resources are created:
+
+    ```sh
+    kubectl get pods,services,configmaps -n signoz
+    ```
+
+    *Check if the required pods, services, and config maps are running and in a healthy state.*
+
+2. Check the logs of the OpenTelemetry Collector:
+
+    ```sh
+    kubectl logs -f -n signoz -l app.kubernetes.io/component=otel-agent
+    ```
+
+    Review the logs for any error messages or indications of misconfiguration.
+
+3. Verify the rendered configuration:
+
+    ```sh
+    kubectl get cm/kubelet-otel-k8s-infra-otel-agent -n signoz -o yaml
+    ```
+
+    Ensure the configuration matches your expected settings and the OpenTelemetry Collector endpoint.
+
+4. Confirm that the necessary Kubernetes resources are created:
+
+    ```sh
+    kubectl get pods,services,configmaps -n platform
+    ```
+
+    Check if the required pods, services, and config maps are running and in a healthy state.
+
+5. Check if all the Pixie deployments are in the correct order:
```

Comment:
Do we need pixie for app tracing?
Maybe we can add 


5. Check logs of signoz:
Depends on how you have installed signoz, check logs of service/ docker container/ pods.

and let Pixie as point 6 if needed!